docs/advanced/virtual-threads.md

```diff
@@ -0,0 +1,74 @@
+## Datafetching threading model without Virtual Threads
+
+Without virtual threads, the DGS framework does not create or manage threads for data fetchers.
+Every data fetcher will run on the same request thread.
+This effectively means that data fetchers do not have any concurrent/parallel behavior, except when explicitly returning `CompletableFuture`.
+The reason for not having concurrent behavior by default is that this would require using a thread per field, which would not scale well.
+
+You can achieve concurrent data fetchers returning `CompletableFuture` from a data fetcher.
+A `CompletableFuture` will let you define the `Executor` it will run on, giving you control over managing thread pools.
+
+## Virtual Threads (JDK 21+)
+
+Working with `CompletableFuture` to get concurrent data fetcher behavior isn't ideal for the developer.
+It adds complexity to the code, and managing and tuning thread pools can be complicated when running at scale.
+
+With the introduction of Virtual Threads in JDK 21 we no longer have to worry about the scalability limitations of creating threads.
+The DGS framework has support for Virtual Threads, which brings concurrent data fetchers out of the box.
+Virtual Thread support can be enabled with the following property:
+
+```properties
+dgs.graphql.virtualthreads.enabled=true
+```
+
+When enabled, each *user defined* data fetcher will run in a new virtual thread.
+Because virtual threads don't need thread pool management, you do not need to configure any thread pool.
+A "user defined data fetcher" is a data fetcher defined with `@DgsQuery`/`@DgsMutation`/`@DgsData`.
+Technically, every field of a POJO also has a data fetcher, but those "simple" data fetchers will not run in separate virtual threads.
+
+Because each data fetcher runs in its own virtual thread, you get concurrent/parallel behavior out of the box.
+
+Note that data fetchers explicitly returning `CompletableFuture` do not get wrapped in a virtual thread.
```

Comment:
Maybe worth highlighting or calling out explicitly  with special formatting so it stands out?